pkg/webhook/stats_reporter.go

```diff
@@ -0,0 +1,127 @@
+package webhook
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+)
+
+const (
+	requestCountName     = "request_count"
+	requestLatenciesName = "request_latencies"
+)
+
+var (
+	requestCountM = stats.Int64(
+		requestCountName,
+		"The number of requests that are routed to webhook",
+		stats.UnitDimensionless)
+	responseTimeInMsecM = stats.Float64(
+		requestLatenciesName,
+		"The response time in milliseconds",
+		stats.UnitMilliseconds)
+
+	requestOperationKey  = tag.MustNewKey("request_operation")
+	kindGroupKey         = tag.MustNewKey("kind_group")
+	kindVersionKey       = tag.MustNewKey("kind_version")
+	kindKindKey          = tag.MustNewKey("kind_kind")
+	resourceGroupKey     = tag.MustNewKey("resource_group")
+	resourceVersionKey   = tag.MustNewKey("resource_version")
+	resourceResourceKey  = tag.MustNewKey("resource_resource")
+	resourceNameKey      = tag.MustNewKey("resource_name")
+	resourceNamespaceKey = tag.MustNewKey("resource_namespace")
+	admissionAllowedKey  = tag.MustNewKey("admission_allowed")
```

Comment:
This is a very useful tag for the admission request, but I am always wary about formatting something as a boolean that might have more than two values now or in the future. In this case, I would suggest tracking three results: allow, deny, and error (from OPA or elsewhere).

Reply:
thanks, replaced admission response with opa response